pkg/controller/elasticsearch/client/model.go

```diff
@@ -44,6 +44,19 @@ type Health struct {
 	ActiveShardsPercentAsNumber float32                  `json:"active_shards_percent_as_number"`
 }
 
+// IsSafeToRoll indicates that a rolling update can continue with the next node if
+// - no relocating or initializing shards or shards being fetched
+// - all primaries allocated
+// only reliable if Health result was created with wait_for_events=languid
+// so that there are no pending initialisations in the task queue
+func (h Health) IsSafeToRoll() bool {
+	return !h.TimedOut && // make sure request did not time out (i.e. no pending events)
+		h.Status != esv1.ElasticsearchRedHealth && // all primaries allocated
+		h.NumberOfInFlightFetch == 0 && // no shards being fetched
+		h.InitializingShards == 0 && // no shards initializing
+		h.RelocatingShards == 0 // no shards relocating
+}
```

Comment:
I think I would not have attached that function to the `Health` struct, and maybe instead move it to a `func IsSafeToRoll(h Health)` close to the rolling upgrade code. It's more a rolling upgrade predicate than something that belongs to the Health domain in my mind.

But that's really not a big deal, I'm fine with keeping the current code.